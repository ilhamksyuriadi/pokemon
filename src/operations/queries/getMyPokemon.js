import { myPokemon } from '../../cache';

const getMyPokemon = () => {
    return myPokemon()
}

export default getMyPokemon