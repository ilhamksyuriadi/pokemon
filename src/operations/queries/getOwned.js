import { gql } from '@apollo/client';

export const GET_OWNED = gql`
    query OwnedNumber{
        poke {
            owned @client
        }
    }
`