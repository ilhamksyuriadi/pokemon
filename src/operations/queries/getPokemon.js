import { gql } from '@apollo/client';

export const GET_POKEMON_DETAIL = gql`
    query pokemon($name: String!) {
        pokemon(name: $name) {
        id
        name
        abilities {
            ability {
                name
            }
        }
        moves {
            move {
                name
            }
        }
            types {
                type {
                    name
                }
            }
        }
    }
`