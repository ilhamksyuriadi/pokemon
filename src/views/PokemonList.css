.list-layout-pokemont-list{
    width: 100vw;
    max-width: 100%;
    min-height: 400px;
    display: flex;
}

.title-pokemont-list{
    margin: 0 0 30px;
}

.button-more{
    font-family: 'Sigmar One', cursive;
    border: none;
    background: black;
    color: white;
    outline: none;
    box-shadow: -2px 2px 0px 0px black;
    border: 1px solid white;
    transition: 0.3s;
    margin: 36px 0;
}

.button-more:hover{
    background: white;
    color: black;
    border: 1px solid black;
    box-shadow: -2px 2px 0px 0px black;
    cursor: pointer;
}

@media (max-width: 500px) {
    .list-layout-pokemont-list{
        flex-direction: column;
        align-items: center;
        min-height: 400px;
    }
}

@media (min-width: 501px) {
    .list-layout-pokemont-list{
        justify-content: space-around;
        align-items: center;
        flex-wrap: wrap;
    }
}