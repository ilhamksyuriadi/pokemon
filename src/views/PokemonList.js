import React from 'react';
import { useQuery } from '@apollo/client';
import { GET_POKEMON_LIST } from '../operations/queries/getPokemonList';
import CardList from '../components/CardList';
import './PokemonList.css';
import Loading from '../components/Loading';
import Error from '../components/Error';
import getMyPokemon from '../operations/queries/getMyPokemon';

const PokemonList = () => {
    const variables = { limit: 8, offset: 0 }
    const {loading, error, data} = useQuery(GET_POKEMON_LIST, { variables })

    if (loading) return <Loading msg='getting pokemon list...' />
    if (error) return <Error />
    if (data) {
        const pokemonList = data.pokemons.results.map((pokemon) => {
            let ownedPokemon = getMyPokemon().filter(({name}) => {
                return pokemon.name === name
            })
            return Object.assign({...pokemon}, {owned: ownedPokemon.length})
        })

        const pokemons = pokemonList.map(({name, image, owned}) => (
            <CardList 
                key={name}
                name={name}
                image={image}
                owned={owned}
            />
        ))

        return (
            <div className="list-layout">
                {pokemons}
            </div>
        )
    }
}

export default PokemonList;