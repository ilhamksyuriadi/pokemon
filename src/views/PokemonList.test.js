import PokemonList from './PokemonList';
import { unmountComponentAtNode } from 'react-dom';
import renderer from  'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import { GET_POKEMON_LIST } from '../operations/queries/getPokemonList';
import { BrowserRouter as Router } from "react-router-dom";

const mocks = [
    {
        request: {
            query: GET_POKEMON_LIST,
            variables: {
                limit: 2,
                offset: 0
            },
        },
        result: {
            data: {
                pokemons: {
                    results: [
                        {
                            name: "bulbasaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                        },
                        {
                            name: "ivysaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
                        },
                        {
                            name: "venusaur",
                            image: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
                        }
                    ]
                }
            }
        },
    },
];

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("integration - pokemon list, snapshot", () => {
    const tree = renderer.create(
        <MockedProvider mock={mocks}>
            <Router>
                <PokemonList />
            </Router>
        </MockedProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})