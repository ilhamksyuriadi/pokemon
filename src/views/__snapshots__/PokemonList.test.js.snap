// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration - pokemon list, snapshot 1`] = `
<div
  className="loading-layout"
>
  <div>
    <img
      alt="pidgey"
      src="pidgey-front.png"
    />
    <h4>
      getting pokemon list...
    </h4>
  </div>
</div>
`;
